distribution/metadata/diff_metadata_service.go

```diff
@@ -0,0 +1,137 @@
+package metadata
+
+import (
+	"encoding/json"
+
+	"github.com/docker/distribution/digest"
+	"github.com/docker/docker/layer"
+)
+
+// DiffMetadataService maps layer IDs to a set of known metadata for
+// the layer.
+type DiffMetadataService struct {
```

Comment:
Should I just go ahead and revert the terminology to BlobSum and BlobSumService and worry about naming in a later release? I'd like to correctly name the metadata on the filesystem if possible